src/core/net/tcp6.cpp

```diff
@@ -37,28 +37,49 @@
 
 #include "tcp6.hpp"
 
+#include <new>
+
 #include "common/code_utils.hpp"
 #include "common/error.hpp"
+#include "common/instance.hpp"
+#include "common/logging.hpp"
+#include "net/ip6.hpp"
 
 namespace ot {
 namespace Ip6 {
 
 Tcp::Tcp(Instance &aInstance)
     : InstanceLocator(aInstance)
+    , mTimer(aInstance, Tcp::HandleTimer)
+    , mFiringTimers(false)
+    , mEphemeralPort(kDynamicPortMin)
 {
+    OT_UNUSED_VARIABLE(mEphemeralPort);
 }
 
 Error Tcp::Endpoint::Initialize(Instance &aInstance, otTcpEndpointInitializeArgs &aArgs)
 {
-    OT_UNUSED_VARIABLE(aInstance);
-    OT_UNUSED_VARIABLE(aArgs);
+    Error error = kErrorNone;
 
-    return kErrorNotImplemented;
+    mContext                  = aArgs.mContext;
+    mEstablishedCallback      = aArgs.mEstablishedCallback;
+    mSendDoneCallback         = aArgs.mSendDoneCallback;
+    mSendReadyCallback        = aArgs.mSendReadyCallback;
+    mReceiveAvailableCallback = aArgs.mReceiveAvailableCallback;
+    mDisconnectedCallback     = aArgs.mDisconnectedCallback;
+
+    mInstance = &aInstance;
```

Comment:
Instead of storing `mInstance` to Endpoint, can we make Endpoint a subclass of InstanceLocator?

Reply:
That wouldn't work, since it corresponds to the `otTcpEndpoint` class in the public API.